Uppercase Shell32 import causes problem when cross compiling on linux

## Summary

I faced issues with the Shell32 library when compiling it via MinGW under Linux.
```
/usr/bin/x86_64-w64-mingw32-ld: cannot find -lShell32: No such file or directory
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

## Fix
I just had to change the name `Shell32` to `shell32` in the following files:
- src/CMakeLists.txt
- tests/CMakeLists.txt

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -51,7 +51,7 @@ if(CMAKE_CXX_STANDARD LESS 14)
 endif()
 
 if(WIN32)
-  target_link_libraries(CLI11 ${PUBLIC_OR_INTERFACE} Shell32)
+  target_link_libraries(CLI11 ${PUBLIC_OR_INTERFACE} shell32)
 endif()
 
 if(CLI11_INSTALL OR CLI11_FULL_INSTALL)

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -163,7 +163,7 @@ function(add_dependent_application_definitions TARGET)
                                PRIVATE CLI11_${APP_UPPERCASE}_EXE="$<TARGET_FILE:${APP}>")
 
     if(WIN32)
-      target_link_libraries(${APP} PRIVATE Shell32)
+      target_link_libraries(${APP} PRIVATE shell32)
     endif()
   endforeach()
 endfunction()

